### PR TITLE
feat(ollama): add configurable thinking mode toggle to settings

### PR DESCRIPTION
**Pull request by eriktews**
Added an option to enable or disable thinking mode for Ollama

- Define ollama_thinking in models.proto and extend OllamaHandlerOptions
- Pass the flag to the Ollama API to control the `think` request parameter
- Add a settings UI checkbox to enable or disable the feature Defaults to enabled to leverage built-in reasoning capabilities when available.

I think this has been discussed in #4628

### Related Issue

No related issues I know about, but there is a discussion:

**Discussion:** #4628

#### What problem does this PR solve?

Right now, thinking is always enabled when using Ollama. That means usually better but also slower answers from the LLM. This PR makes it possible to disable thinking.

#### Why were these changes introduced and what purpose do they serve?

Improve performance for Ollama users.

#### For larger changes, provide context about your approach and reasoning

That's a very small change

### Test Procedure

Tested it locally with qwen3.6:35b and verified that the requests go through as planned using a network sniffer. By default, thinking is enabled, which should not break anything.

### Type of Change

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [X] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

-   [X] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [ ] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [X] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

Tests are not passing due to an unrelated problem currently in the main branch. The current code in the main branch is also not properly formatted, at least I get errors when I run the linter there.

### Screenshots

Bildschirmfoto vom 2026-06-07 18-29-26

### Additional Notes

Feel free to reformat this PR.

### CODE REVIEW

**Comment by eriktews** [comment]
Here is an updated screenshot of the new version in 457bff114b84b8d7dde3cb998ec6a55bd7b5d3d4
Bildschirmfoto vom 2026-06-07 19-48-17

File: apps/vscode/proto/cline/models.proto [hunk 1]
```diff
@@ -600,6 +600,7 @@ message ModelsApiConfiguration {
   optional string nous_research_api_key = 85;
   optional bool azure_identity = 86;
   optional string wandb_api_key = 87;
+  optional bool ollama_thinking = 88;
```
[the remark this answers is not in the document]
Comment (eriktews) [reply apps/vscode/proto/cline/models.proto]: This has been fixed in 457bff114b84b8d7dde3cb998ec6a55bd7b5d3d4

[the remark this answers is not in the document]
Comment (eriktews) [reply apps/vscode/proto/cline/models.proto]: Sorry, this specific bug is still there in 457bff114b84b8d7dde3cb998ec6a55bd7b5d3d4

[the remark this answers is not in the document]
Comment (eriktews) [reply apps/vscode/proto/cline/models.proto]: This should be fixed in 5626f1594a4fbd02da72d90ccea47e95963d9498

**Comment by eriktews** [comment]
That should fix all issues addressed by the CI pipeline, please review the request manually now and approve it.

**Review by dominiccooney — CHANGES_REQUESTED** [review]
The user-facing toggle is still useful, but the implementation needs to move to the current SDK provider path.

#10678 already added the approved explicit-disable routing for Ollama models whose reasoning defaults on. Please retarget this PR to the current `OllamaProvider` and persist the selection through the existing `ProviderSettings.reasoning` field (`useProviderConfig("ollama").write(...)`). That keeps VS Code and CLI on the same source of truth and lets `cline-session-factory` carry the setting into the SDK request. Please do not add a second Ollama-specific proto/global-state field.

For this PR, keep the control to the behavior supported by the linked discussion and current routing contract: Default (unset), Enabled, and Disabled. The later `high`/`medium`/`low` expansion needs separate model capability and native-adapter support before the UI can promise it for dynamically discovered Ollama models.

Please add tests that cross the real boundaries:

- the Ollama settings control reads, writes, and clears `ProviderSettings.reasoning`;
- reloading the provider config preserves each selection;
- the session factory maps enabled/disabled settings coherently;
- the existing Ollama routing tests prove Disabled emits the provider wire patch while Default remains omitted.

Thank you for driving this from #4628. The SDK migration did most of the backend work, so the remaining current-main change should be a much smaller settings integration.

**Comment by eriktews** [comment]
As far as I know, there is no way to figure out whether a model in Ollama supports low/medium/high or just true/false. I think we have two options here:

* Always offer all 5 settings, hoping that the user will pick a good one (that's what I prefer)
* Do some string matching on the model name but that will definitely not work everywhere and break when a user uses a less common model or renames his model

What would you suggest?
